src/tests/ftest/server/daos_server_restart.py

```diff
@@ -0,0 +1,218 @@
+            force (bool): Force to stop the daos io_server.
+            Defaults to True.
+
+        """
+        try:
+            self.server_managers[0].dmg.system_stop(force)
+        except CommandFailure as error:
+            self.fail(
+                "##Failed stopping daos io-server by dmg, %s", error)
+        try:
+            self.server_managers[0].dmg.system_start()
+        except CommandFailure as error:
+            self.fail(
+                "##Failed starting daos io-server by dmg, %s", error)
+
+    def get_pool_list(self):
+        """method to get the pool list contents"""
+        pool_list = self.get_dmg_command().get_output("pool_list")
+        self.log.info("get_pool-list: %s", pool_list)
+        return pool_list
+
+    def verify_pool_list(self, expected_pool_list=None):
+        """method to verify the pool list"""
+        if expected_pool_list is None:
+            expected_pool_list = []
+        pool_list = self.get_pool_list()
+        self.log.info(
+            "\n===Current pool-list:  %s\n===Expected pool-list: %s\n",
+            pool_list, expected_pool_list)
+        self.assertEqual(
+            pool_list, expected_pool_list,
+            "##Current pool-list mismatch with the expected pool-list.")
+
+    def create_pool_and_container(self):
+        """method to create pool and container"""
+        scm_size = self.params.get("scm_size", "/run/server/*/", 138000000)
+        num_of_pool = self.params.get("num_of_pool", "/run/server/*/", 3)
+        container_per_pool = self.params.get(
+            "container_per_pool", "/run/server/*/", 2)
+        for _ in range(num_of_pool):
+            dmg = self.get_dmg_command()
+            result = dmg.pool_create(scm_size)
+            uuid = result['uuid']
+            svc = result['svc']
+            for _ in range(container_per_pool):
+                daos_cmd = DaosCommand(self.bin)
+                daos_cmd.exit_status_exception = False
+                result = daos_cmd.container_create(pool=uuid, svc=svc)
+                self.log.info("container create status: %s", result)
+
+    def test_daos_server_reformat(self):
+        """JIRA ID: DAOS-3596.
+
+        Test Description: verify reformatting DAOS is same as a fresh install.
+        Steps:
+        (1)Verify daos server pool list after started.
+        (2)Restart server without pool been created, and verify.
+        (3)Create pools, containers and verify the pools.
+        (4)Perform an orderly DAOS shutdown, and Use cmd line tools to
+           re-format both SSDs and NVDIMMs.
+        (5)Verify after DAOS server restarted, it should appear as an empty
+           fresh installation.
+
+        :avocado: tags=all,pr,hw,large,server_test,server_reformat
+        """
+
+        self.log.info("(1)Verify daos server pool list after started.")
+        self.verify_pool_list()
+        self.log.info("(2)Restart server without pool created and verify.")
+        self.restart_daos_server()
```

Comment:
This will only restart 1 server is that what is supposed to happen?

Reply:
same as above.